File: go-pr/48/go/code-quality-naming/network_handler.go
package main

import "fmt"

type NetworkHandler struct {
	// {fact rule=code-quality-naming@v1.0 defects=1}
	url     string
	port    int
	conn    bool
	timeout int
	// {/fact}
	
	// {fact rule=code-quality-naming@v1.0 defects=0}
	serverURL       string
	serverPort      int
	isConnected     bool
	timeoutDuration int
	// {/fact}
}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (nh *NetworkHandler) connect(u string, p int) bool {
	nh.url = u
	nh.port = p
	nh.conn = true
	return true
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (nh *NetworkHandler) EstablishConnection(serverURL string, serverPort int) bool {
	nh.serverURL = serverURL
	nh.serverPort = serverPort
	nh.isConnected = true
	return true
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (nh *NetworkHandler) send(data string) string {
	return "OK"
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (nh *NetworkHandler) SendDataToServer(requestData string) string {
	return "OK"
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (nh *NetworkHandler) close() {
	nh.conn = false
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (nh *NetworkHandler) CloseConnection() {
	nh.isConnected = false
}
// {/fact}

func main() {
	handler := &NetworkHandler{}
	handler.connect("http://api.example.com", 8080)
	response := handler.send("test data")
	fmt.Println(response)
	handler.close()
}

File: ts-pr/55/typescript/code-quality-naming/network_handler.ts
class NetworkHandler {
    // {fact rule=code-quality-naming@v1.0 defects=1}
    private url: string = '';
    private port: number = 0;
    private conn: boolean = false;
    private timeout: number = 0;
    // {/fact}
    
    // {fact rule=code-quality-naming@v1.0 defects=0}
    private serverURL: string = '';
    private serverPort: number = 0;
    private isConnected: boolean = false;
    private timeoutDuration: number = 0;
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    connect(u: string, p: number): boolean {
        this.url = u;
        this.port = p;
        this.conn = true;
        return true;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    establishConnection(serverURL: string, serverPort: number): boolean {
        this.serverURL = serverURL;
        this.serverPort = serverPort;
        this.isConnected = true;
        return true;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    send(data: string): string {
        return 'OK';
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    sendDataToServer(requestData: string): string {
        return 'OK';
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    close(): void {
        this.conn = false;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    closeConnection(): void {
        this.isConnected = false;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    get st(): boolean {
        return this.conn;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    getConnectionStatus(): boolean {
        return this.isConnected;
    }
    // {/fact}
}

const handler = new NetworkHandler();
handler.connect('http://api.example.com', 8080);
const response = handler.send('test data');
console.log(response);
handler.close();